### PR TITLE
Find command bugs v1.4

### LINKED ISSUE #131: [PE-D] Find command accepts random inputs
The ```find``` command appears to be accepting random (and most likely invalid) params and lists everyone. It would be better to return nothing as the query result, or warn the user that inputs are invalid.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: zhuoyang125/ped#2

### LINKED ISSUE #145: [PE-D] unintended interaction of find command?
when I use the command `find somethingsomething`, it does the same thing as `list` and displays everyone, there is also no error message, not sure if this is meant to be part of the condition based filter

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: syoopie/ped#8

### LINKED ISSUE #149: [PE-D] `find` command allows invalid inputs
when command `find charlotte` is used without the `\n` tag for name, find command does not throw error, instead it shows the entire list.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: marcuschj/ped#8

### LINKED ISSUE #166: [PE-D] Wrong command result message
Issue:

Upon executing `clear`, the response is 

instead of something like `Heron has been cleared!`

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: luodan01/ped#2

### LINKED ISSUE #173: [PE-D] Invalid keywords in Find Command not handled
Not sure if this can be considered a functionality or documentation bug, but searching for invalid keywords do not seem to do anything. For example, the following command does not update the list. If the user can't remember all the valid keywords, the following command gives the false impression that all employees are considered `sksksk` in the system

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: luodan01/ped#6

### PR DESCRIPTION
**Pull request by beefham**
Fix situation where random words are allowed as a preamble to the find command. Now only accepts `find unpaid`. (Resolves #131 , resolves #145, resolves #149, resolves #173)
Also change clear command's success message to mention HeRon. (Resolves #166)

### CODE REVIEW

**Review by 1waykiat — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -77,16 +79,23 @@ public FindCommand parse(String args) throws ParseException {
...
+            // no arguments provided to the command
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+        if (!preamble.isBlank()) { // a preamble exists
+            if (!preamble.equals(UNPAID_PREDICATE_KEYWORD)) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+        }
```
Comment (1waykiat) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:93]:
```suggestion
        if (!preamble.isBlank() && !preamble.equals(UNPAID_PREDICATE_KEYWORD) ) {  // preamble that is not 'unpaid'
                throw new ParseException(
                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
        }
```
Perhaps consider merging both tests together?

**Review by 1waykiat — COMMENTED** [review]
Was looking through the code again and thought of some improvements again, might want to consider them.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (beefham) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java:93]: Yup that sounds good!

**Pull request merged**